jcl/src/java.base/share/classes/java/lang/invoke/CollectHandle.java

```diff
@@ -99,10 +139,10 @@ protected final ThunkTuple computeThunks(Object arg) {
 		return thunkTable().get(new ThunkKeyWithIntArray(ThunkKey.computeThunkableType(type()), collectArguments));
 	}
 
-	private final Object allocateArray() {
+	private static final Object allocateArray(int size, Class<?> clazz) {
```

Comment:
Is this the majority of the improvement?  Moving the deref to the ctor so it becomes a field load at runtime?